src/forces.f90

```diff
@@ -256,267 +268,626 @@ subroutine force(ux1,uy1,ep1)
+         end if
+         call MPI_ALLREDUCE(tdiffyl,tdiffy,nz,real_type,MPI_SUM,MPI_COMM_WORLD,code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+
+         do k=1,zsize(3)
+
+            tpresx(k)=tpresx(k)/dt
+            tpresy(k)=tpresy(k)/dt
+
+            xmom    = tunstx(k)+tconvx(k)
+            ymom    = tunsty(k)+tconvy(k)
+            xDrag(k) = two*(tdiffx(k)+tpresx(k)-xmom)
+            yLift(k) = two*(tdiffy(k)+tpresy(k)-ymom)
+
+         enddo
 
-       !*****************************************************************
-       !        Calculation of the momentum terms
-       !*****************************************************************
-       !
-       !     Calculation of the momentum terms. First we integrate the 
-       !     time rate of momentum along the CV.
-       !
-       !         Excluding the body internal cells. If the centroid 
-       !         of the cell falls inside the body the cell is 
-       !         excluded.
-
-       tunstxl=zero
-       tunstyl=zero
-       do k=1,xsize(3)
-          tsumx=zero
-          tsumy=zero
-          do j=jcvlw_lx(iv),jcvup_lx(iv)
-             do i=icvlf_lx(iv),icvrt_lx(iv)
-                !     The velocity time rate has to be relative to the cell center, 
-                !     and not to the nodes, because, here, we have an integral 
-                !     relative to the volume, and, therefore, this has a sense 
-                !     of a "source".
-                !         fac   = (1.5*ux1(i,j,k)-2.0*ux01(i,j,k)+0.5*ux11(i,j,k))*epcv1(i,j,k)
-                fac   = (onepfive*ux1(i,j,k)-two*ux01(i,j,k)+half*ux11(i,j,k))*(one-ep1(i,j,k))
-                tsumx = tsumx+fac*dx*del_y(j+(xstart(2)-1))/dt    !tsumx+fac*dx*dy/dt
-                !sumx(k) = sumx(k)+dudt1*dx*dy
-
-                !         fac   = (1.5*uy1(i,j,k)-2.0*uy01(i,j,k)+0.5*uy11(i,j,k))*epcv1(i,j,k)
-                fac   = (onepfive*uy1(i,j,k)-two*uy01(i,j,k)+half*uy11(i,j,k))*(one-ep1(i,j,k))
-                tsumy = tsumy+fac*dx*del_y(j+(xstart(2)-1))/dt !tsumy+fac*dx*dy/dt
-                !sumy(k) = sumy(k)+dudt1*dx*dy
+         !Edited by F. Schuch
+         xDrag_mean = sum(xDrag(:))/real(nz,mytype)
+         yLift_mean = sum(yLift(:))/real(nz,mytype)
+
+         !     if ((itime==ifirst).or.(itime==0)) then
+         !        if (nrank .eq. 0) then
+         !        write(filename,"('aerof',I1.1)") iv
+         !        open(38+(iv-1),file=filename,status='unknown',form='formatted')
+         !        endif
+         !     endif
+         if (nrank .eq. 0) then
+            write(38,*) t,xDrag_mean,yLift_mean
+            flush(38)
+         endif
+         if (mod(itime, icheckpoint).eq.0) then
+            if (nrank .eq. 0) then
+               write(filename,"('forces.dat',I7.7)") itime
+               call execute_command_line ("cp forces.dat "//filename)
```

Comment:
Is it possible to perform this `cp` operation only once after the `if i2dsim` / `elseif i2dsim` / `endif` ?

Currently this operation is performed multiple times when `nvol > 1`

Reply:
Will check if it is possible.